test/cmd/util.sh

```diff
@@ -0,0 +1,232 @@
+#!/bin/bash
+# This utility file contains functions that wrap commands to be tested
+# and allow us to interface our command-line tests with JUnit on Jenkins.
+# All wrapper functions run commands in a sub-shell and redirect all output.
+# Tests in test-cmd *must* use these functions for testing.
+
+# expect_exit_code_and_text runs the provided test command and expects a 
+# specific exit code from that command as well as the success of a specified
+# `grep` invocation. Output from the command to be tested is suppressed unless
+# either `VERBOSE=1` or the test fails.
+# this function expects to have ${JUNIT_OUTPUT_FILE} set in order to direct
+# jUnit-intended output to that file instead of stdout
+function expect_exit_code_run_grep() {
+	cmd=$1
+	# default expected cmd code to 0 for success
+	expected_cmd_code=${2:-'0'}
+	# default grep args to a something that always passes, even for zero-length input
+	grep_args=${3:-'$'} 
+	# default expected test code to 0 for success
+	expected_test_code=${4:-'0'}
+
+	local startTime endTime timeElapsed
+
+ 	declare_test_start "${cmd}" >> ${JUNIT_OUTPUT_FILE}
```

Comment:
Not unique enough.  Is there a way to automatically harvest the line number of origin?